assets/js/modules/tagmanager/setup.js

```diff
@@ -297,72 +350,80 @@ class TagmanagerSetup extends Component {
 
 			global.googlesitekit.modules.tagmanager.settings = savedSettings;
 
-			if ( this._isMounted ) {
-				this.setState( {
-					isSaving: false,
-				} );
-			}
+			this.setState( {
+				isSaving: false,
+			} );
 		} catch ( err ) {
-			if ( this._isMounted ) {
-				this.setState( {
-					isLoading: false,
-					errorCode: err.code,
-					errorMsg: err.message,
-				} );
-			}
+			this.setState( {
+				isLoading: false,
+				isSaving: false,
+				errorCode: err.code,
+				errorMsg: err.message,
+			} );
 
-			// Catches error in handleButtonAction from <SettingsModules> component.
-			return new Promise.reject( err );
+			// Re-throw the error to return a rejected promise.
+			throw err;
 		}
 	}
 
 	static createNewAccount( e ) {
 		e.preventDefault();
-		global.open( 'https://marketingplatform.google.com/about/tag-manager/', '_blank' );
+		global.open( 'https://tagmanager.google.com/#/admin/accounts/create', '_blank' );
 	}
 
 	handleAccountChange( index, item ) {
 		const { selectedAccount } = this.state;
-		const selectValue = item.getAttribute( 'data-value' );
+		const selectValue = item.dataset.value;
 
 		if ( selectValue === selectedAccount ) {
 			return;
 		}
 
-		if ( this._isMounted ) {
+		this.setState( { selectedAccount: selectValue } );
+
+		if ( ACCOUNT_CHOOSE === selectValue ) {
 			this.setState( {
-				containersLoading: true,
-				selectedAccount: selectValue,
+				selectedContainer: ACCOUNT_CHOOSE,
+				containers: [ {
+					publicId: ACCOUNT_CHOOSE,
+					name: __( 'Select an account', 'google-site-kit' ),
```

Comment:
This should say "Select a container". Having it say something different just because you haven't yet interacted with the other dropdown is confusing - the field being disabled is the only thing that should be different.

Reply:
This was borrowed from Analytics setup for consistency but that makes sense.

Reply:
When an option value is empty, the enhanced select only displays its label, even if there is an option with a matching value.

![image](https://user-images.githubusercontent.com/1621608/73844358-1916b900-4829-11ea-9299-001540da00da.png)

What do you think about using this instead of adding placeholder options? This seems to be more consistent with MDC behavior and results in less translation strings but maybe not as good for accessibility?

Reply:
I would go for the better accessibility. Having these placeholder options would also work in a regular `select`, so I think we should go for that.